arquillian/api/src/main/java/org/wildfly/swarm/arquillian/adapter/UberjarSimpleContainer.java

```diff
@@ -130,7 +130,15 @@ public void start(Archive<?> archive) throws Exception {
                 .fractionDetectionMode(BuildTool.FractionDetectionMode.never)
                 .bundleDependencies(false);
 
-        final String additionalModules = System.getProperty(SwarmInternalProperties.BUILD_MODULES);
+        String additionalModules = System.getProperty(SwarmInternalProperties.BUILD_MODULES);
+
+        // See https://issues.jboss.org/browse/SWARM-571
+        if(null==additionalModules) {
+            // see if we can find it
+            File modulesDir = new File("target/classes/modules");
+            additionalModules = modulesDir.exists() ? modulesDir.getAbsolutePath() : null;
+        }
+
```

Comment:
I was thinking about that as well. But that would mean we need to prevent duplicate locations to be added through both means. Therefore I've decided to make it mutual exclusive for now.